core/src/test/java/org/infinispan/configuration/CustomInterceptorInjectionTest.java

```diff
@@ -16,14 +25,29 @@
 
    @Override
    protected EmbeddedCacheManager createCacheManager() throws Exception {
-      ConfigurationBuilder c = getDefaultStandaloneCacheConfig(false);
-      c.customInterceptors().addInterceptor().index(0).interceptor(new SomeInterceptor());
-      return TestCacheManagerFactory.createCacheManager(c);
+      ConfigurationBuilder builder = getDefaultStandaloneCacheConfig(false);
+      builder.customInterceptors().addInterceptor().index(0).interceptor(new SomeInterceptor());
+      return TestCacheManagerFactory.createCacheManager(builder);
+   }
+
+   public void testInjection() {
+      AsyncInterceptor interceptor = cache.getAdvancedCache().getAsyncInterceptorChain().getInterceptors().get(0);
+      assertEquals(SomeInterceptor.class, interceptor.getClass());
+      SomeInterceptor someInterceptor = (SomeInterceptor) interceptor;
+      assertSame(cache.getAdvancedCache().getLockManager(), someInterceptor.lm);
+      assertSame(cache.getAdvancedCache().getDataContainer(), someInterceptor.dc);
    }
 
-   public void testInjectionWorks() {
-      final Cache<Object,Object> cache1 = cacheManager.getCache();
-      assert cache1.getAdvancedCache().getInterceptorChain().get(0).getClass().equals(SomeInterceptor.class);
-      assert SomeInterceptor.lm != null;
+   static class SomeInterceptor extends BaseCustomAsyncInterceptor {
```

Comment:
I think we should keep the `CommandInterceptor`-based test as well, for now.

Reply:
done.